### PR TITLE
Kernels generation: replace datasynth with the synthesis module

### PR DESCRIPTION
**Pull request by nightwnvol**
Added the **synthesis.py** module to generate kernels without using the **datasynth** command

### CODE REVIEW

File: amico/models.py [hunk 1]
```diff
@@ -1,18 +1,13 @@
...
-import tempfile
+from os.path import join as pjoin
 import amico.lut
 from tqdm import tqdm
-from dipy.core.gradients import gradient_table
-from dipy.sims.voxel import single_tensor
 import abc
-from amico.util import PRINT, LOG, NOTE, WARNING, ERROR, get_verbose
+from amico.util import PRINT, ERROR, get_verbose
+from amico.synthesis import Stick, Zeppelin, Ball, CylinderGPD, SphereGPD, Astrosticks, NODDISignal
```
Comment (daducci) [inline amico/models.py:10]: Can we find a better alternative for the `NODDIsignal` name?

File: amico/models.py [hunk 2]
```diff
@@ -231,27 +226,30 @@ def set_solver( self ) :
...
+        stick = Stick(scheme=scheme_high)
+        zeppelin = Zeppelin(scheme=scheme_high)
+        ball = Ball(scheme=scheme_high)
 
         nATOMS = 1 + len(self.d_perps_zep) + len(self.d_isos)
         idx = 0
         with tqdm(total=nATOMS, ncols=70, bar_format='   |{bar}| {percentage:4.1f}%', disable=(get_verbose()<3)) as progress:
             # Stick
-            signal = single_tensor( gtab, evals=[self.d_perp, self.d_perp, self.d_par] )
+            signal = stick.get_signal(diff=self.d_par, theta=0, phi=0)
```
Comment (daducci) [inline amico/models.py:238]: For all models, can we define the two parameters `theta` and `phi` to default to 0, so we don't need to specify them in the calls? At least, this is true in this specific example, but of course someone else might want to use these functions to generate signals in different orientations

File: amico/synthesis.py [hunk 3]
```diff
@@ -0,0 +1,638 @@
+import numpy as np
+import scipy
+from amico.util import ERROR
+
+# Limits the required precision in gpd sum
+REQUIRED_PRECISION = 1e-7
+
+# Gyromagnetic ratio
+# GAMMA = 2.675153151e8 # https://physics.nist.gov/cgi-bin/cuu/Results?search_for=proton+gyromagnetic+ratio
```
Comment (daducci) [inline amico/synthesis.py:9]: Remove unnecessary comments (everywhere, let's clean up the code)

File: amico/synthesis.py [hunk 4]
```diff
@@ -0,0 +1,638 @@
...
+        
+        self.b = self.scheme.b
+
+    def get_signal(self, diff):
+        signal = np.exp(-self.b * diff)
+        return signal
+
+# CYLINDER
+class CylinderGPD():
+    AM_CYLINDER = np.array([
```
Comment (daducci) [inline amico/synthesis.py:157]: underscored?

File: amico/synthesis.py [hunk 5]
```diff
@@ -0,0 +1,638 @@
...
+                l_perp = 0
+                l_par = -b / (g * g) * diff
+                signal[i] = np.sqrt(np.pi) * 1 / (2 * g * np.sqrt(l_perp - l_par)) * np.exp(g * g * l_perp) * scipy.special.erf(g * np.sqrt(l_perp - l_par))
+        return signal
+
+# BALL
+class Ball():
+    def __init__(self, scheme):
+        self.scheme = scheme
+
```
Comment (daducci) [inline amico/synthesis.py:148]: Remove unnecessary spaces/empty lines

File: amico/synthesis.py [hunk 6]
```diff
@@ -0,0 +1,638 @@
...
+    last_small_delta = 0.0
+    last_diff = 0.0
+    last_radius = 0.0
+    last_sum = 0.0
+
+    def __init__(self, scheme):
+        self.scheme = scheme
+
+    def get_signal(self, diff, radius):
+        diff *= 1e-6
```
Comment (daducci) [inline amico/synthesis.py:72]: Is this necessary or we can assume always a given units? Before there was the problem with datasynth, but now all models should use the very same convention, no?

File: amico/models.py (hunk 1 above)
Comment (nightwnvol) [reply amico/models.py:10]: What about three separate classes for each NODDI compartment?
e.g. `NODDIIntraCellular()`, `NODDIExtraCellular()` and `NODDIIsotropic`

Comment (daducci) [reply amico/models.py:10]: This can be a suitable option!

Comment (nightwnvol) [reply amico/models.py]: commit 733ab16ad6e050ca44ba56e8cd154c82c45de922

**Pull request merged**